src/main.js

```diff
@@ -145,6 +147,35 @@ class PercyClient {
     return this._httpPost(`${this.apiUrl}/builds/${buildId}/resources/`, data);
   }
 
+  uploadResources(buildId, contents) {
```

Comment:
Can you make this accept a `resources` arg instead of contents? We actually have two states for a Resource object, it can either have sha && content on it, or it can have sha && localPath on it — which then the upload function should read `content` from `localPath` if that is set instead. This allows us to avoid holding all contents in memory 100% of the time (for some SDKs).  We always use the localPath method in ember-percy, for example: https://github.com/percy/ember-percy/blob/master/index.js#L250

Reply:
Just to clarify, are you wanting me to bake that logic in here - given a resource object with a `localPath` rather than `sha`, read it from disk and then upload it?

Reply:
Yes please, I think the arguments here should be `buildId, resources` and then you do something like:

```js
let content =  resource.localPath ? fs.readFileSync(resource.localPath) : resource.content
```
(or the equivalent if statement if it's easier to read)

Reply:
The `.sha` will always always exist, not related here. But we either require `localPath` OR `content` to exist.

Reply:
Done!